File: IronVault/IronVault.Model/Models/Recenzija.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace IronVault.Model.Models
{
    public partial class Recenzija
    {
        public int RecenzijaId { get; set; }
        public int KorisnikId { get; set; }
        public int SuplementId { get; set; }
        public int Ocjena { get; set; }
        public string Opis { get; set; } = null!;
        public virtual Korisnik Korisnik { get; set; } = null!;
        public virtual Suplement Suplement { get; set; } = null!;
    }
}

File: IronVault/UI/ironvault_mobile/lib/models/recenzija.dart
import 'package:ironvault_mobile/models/korisnik.dart';
import 'package:ironvault_mobile/models/suplement.dart';
import 'package:json_annotation/json_annotation.dart';

part 'recenzija.g.dart';

@JsonSerializable()
class Recenzija {
  int? recenzijaId;
  int? korisnikId;
  Korisnik? korisnik;
  int? suplementId;
  Suplement? suplement;
  int? ocjena;
  String? opis;

  Recenzija({this.recenzijaId});

  factory Recenzija.fromJson(Map<String, dynamic> json) =>
      _$RecenzijaFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$RecenzijaToJson(this);
}
